src/OrchardCore/OrchardCore.Abstractions/Json/Dynamic/JsonDynamicObject.cs

```diff
@@ -3,17 +3,18 @@
 using System.Dynamic;
 using System.Reflection;
 using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
 using System.Text.Json.Settings;
 
 #nullable enable
 
 namespace System.Text.Json.Dynamic;
 
 [DebuggerDisplay("JsonDynamicObject[{Count}]")]
-public class JsonDynamicObject : DynamicObject
+[JsonConverter(typeof(JsonDynamicJsonConverter<JsonDynamicObject>))]
+public class JsonDynamicObject : JsonDynamicBase
```

Comment:
Can you make all these classes sealed (for perf)? Or do we expect users to be able to sub-class it? If you have some hesitation just use sealed, and we can unseal it later if someone comes with a real use-case.

Reply:
Fair point, I don't think they should be able to sub-class. I'll seal them all.

Reply:
Done.